### PR TITLE
Added push notifications

### PR DESCRIPTION
**Pull request by willsr**
Added push notifications to the library.

### CODE REVIEW

File: parse.php [hunk 1]
```diff
@@ -85,6 +101,31 @@ public function create($args){
...
+ * 
+ * @param array $args - argument hash:
+ * 
+ * push: leave this alone
+ * object: notification details
+ * 
+ * @return string $return
+ * 
+ */
+	public function notification($args){
```
Comment (willsr) [inline parse.php:115]: could include a more elegant solution that handles the data class nested within the payload "object"

**Comment by andrewscofield** [comment]
Hey, this looks great, thanks! I will have more time later this weekend to play around with it, but at a quick glance it looks like the direction I was going to be heading anyway for adding user, file and geopoint support.

File: parse.php [hunk 2]
```diff
@@ -85,6 +101,31 @@ public function create($args){
...
+		$params = array(
+			'url' => 'push',
+			'method' => 'POST',
+			'payload' => $args['object']
+		);
+		
+		$return = $this->request($params);
+
+		return $this->checkResponse($return,'201');
```
Comment (MuzzleFork) [inline parse.php:124]: Should this be checking for status 200? It seems to return status 200 locally, with {"result":true} and the push notification does get sent. Using 201 makes this return a false error.

**Comment by kasperfn** [comment]
Additional comment and code: 
When updating 'Channels' for a device, we don't look at "/1/classes/installations" but "/1/installations". So we need pushUrl and not parseUrl here.

in update($args) or updateChannel($args)

```
    $params = array(
        'url' => $args['className'].'/'.$args['objectId'],
        'className' => 'installations',
        'method' => 'PUT',
        'payload' => $args['payload']
    );
```

in request($args):

```
    if( ($args['url'] == "push")  ) {
        curl_setopt($c, CURLOPT_URL, $this->pushUrl . $args['url']); // ../
    }else if( (isset($args['className']) && $args['className'] == "installations") ) {
        curl_setopt($c, CURLOPT_URL, $this->pushUrl . $args['url'] ); // ../
    } else {
        curl_setopt($c, CURLOPT_URL, $this->parseUrl . $args['url']); // ../classes/..
    }
```

It was a quick fix for my case :)

**Comment by andrewscofield** [comment]
Good call! Thanks for sending this fix too!

**Pull request merged**